File: Factorial.scala
object Factorial{

  def factorial(x: BigInt): BigInt = 
    if(x == 0) 1 else x * factorial(x -1)

  def main(args: Array[String])={
    println("factorial(5) = " + factorial(5))
    println("factorial(30) = " + factorial(30))
  }
}

File: Factorial.pl
#!/usr/bin/perl -w
sub factorial {
	$f = 1; 
	$n = $t = $_[0];
	for($i = 0; $i < $t; $i++)
	{
		$f = $f*$n;
		$n = $n-1;
	}
	return $f;
}
print "factorial(5) = " . factorial(5) . "\n";
print "factorial(30) = " . factorial(30) . "\n";

